service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/UploadC2DocumentsController.java

```diff
@@ -49,13 +54,23 @@ public AboutToStartOrSubmitCallbackResponse handleAboutToSubmit(
         @RequestHeader(value = "authorization") String authorization) {
         Map<String, Object> data = callbackrequest.getCaseDetails().getData();
         CaseData caseData = mapper.convertValue(data, CaseData.class);
-        
+
         data.put("c2DocumentBundle", buildC2DocumentBundle(caseData, authorization));
         data.remove("temporaryC2Document");
 
         return AboutToStartOrSubmitCallbackResponse.builder().data(data).build();
     }
 
+    @PostMapping("/submitted")
+    public void handleSubmittedEvent(
+        @RequestHeader(value = "authorization") String authorization,
+        @RequestHeader(value = "user-id") String userId,
+        @RequestBody CallbackRequest callbackRequest) {
+
+        applicationEventPublisher.publishEvent(new C2UploadNotifyEvent(callbackRequest, authorization, userId,
```

Comment:
I am not fan of the name `C2UploadNotifyEvent`. This should know nothing about any notification being send. The event should simply state what happened, i.e. C2Uploaded.

Reply:
updated.... naming can be hard 😄